### ISSUE TITLE
No way to commit one message at a time when many are in the queue

### ISSUE THREAD

**Issue by ubershmekel** [nodefluent/node-sinek#7]
Hi @krystianity it seems I'm losing data. I was hoping to use node-sinek after reading this statement:

>auto-commit / manual drain commit in backpressure-mode (dont loose data)

My use-case is a minute-long process for every message. It's important for me to commit every time I finish processing a single message and not before I'm done processing. Currently because of Drainer._q I see I'm buffering tens of messages and the queue is only drained and committed after many are processed. If I try to `consumer.consumer._commit` then all of my queued messages are committed.

**Comment by krystianity** [comment]
Hi @ubershmekel you are completely right, currently (even in backpressure mode) grab a buffer full of messages, stop the process, and commit the partitions states when the batch is processed, right before continuing to fetch for the next buffer.

From your comments (opened issues) at kafka-node I see that you have already realised that we simply implemented the suggested backpressure method here. However I am still interested in offering a one-by-one commit api for sinek.

From previous attempts with kafka-node I already made the experience that using the offset class to commit specific partition offsets manually can be tricky and prone to bugs.

Since you already switched back to rabbit-mq for your task, I wont handle this with any priority yet I might add the feature in the near future.
